src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.Business;
+import com.google.sps.MockDataBusiness;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@WebServlet("/search")
+public class SearchServlet extends HttpServlet {
+
+    Gson gson = new Gson();
+    MockDataBusiness mockDataBusiness = new MockDataBusiness();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        handleRequest(request, response);
+    }
+
+    public void handleRequest(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        String resultJson;
+        resp.setContentType("application/json;");
+
+        // Get Parameters From The Request
+        String searchQuery = req.getParameter("s");
+        if (searchQuery == null || "".equals(searchQuery)) {
+            resultJson = gson.toJson(mockDataBusiness.mockDataList);
+        } else {
+            List<String> words = Arrays.asList(searchQuery.toLowerCase().split(" "));
+            List<Business> searchedResult = mockDataBusiness.mockDataList
+                    .stream()
+                    .filter(business -> words.stream()
+                    .anyMatch(word ->
+                            business.getBusinessName().toLowerCase().contains(word)))
+                    .collect(Collectors.toList());
+            resultJson = gson.toJson(searchedResult);
```

Comment:
Consider converting the streaming to use the [Datastore's built-in filter search](https://cloud.google.com/datastore/docs/concepts/queries#query_interface), rather than streaming and filtering after fetching the whole list of businesses. This won't scale if we have many businesses, and makes us fetch the entire business database on every search.

Reply:
Will you be changing this? If this is something you wanna do in the future, you can create an issue, let me know if you're gonna work on this now or not, so i can approve this pr!

Reply:
@jacoblipech

Reply:
okies, I will do this in the future PR since I have to change the way this function populates data from entity instead of mockData.